kernel/aster-nix/src/fs/ext2/inode.rs

```diff
@@ -1551,31 +1609,39 @@ impl InodeImpl {
     }
 
     pub fn set_device_id(&self, device_id: u64) {
-        self.0.write().desc.block_ptrs.as_bytes_mut()[..core::mem::size_of::<u64>()]
+        let mut inner = self.0.write();
+        inner.desc.block_ptrs.as_bytes_mut()[..core::mem::size_of::<u64>()]
             .copy_from_slice(device_id.as_bytes());
+        inner.desc.mtime = inner.current_time();
```

Comment:
I don't think `set_device_id` is a good location to update the mtime.

Currently, the only user of `set_device_id` is `mknod`. A newly-created inode returned from `self.create` already has its timestamps up-to-date. There is no point of updating it in `set_device_id` again immediately after `self.create`. In fact, I don't even know why we should have a `set_device_id` method. The device ID should have been determined _when_ the inode is created, _not after_ it is created.

```rust
    fn mknod(&self, name: &str, mode: InodeMode, dev: Arc<dyn Device>) -> Result<Arc<dyn Inode>> {
        let inode = self.create(name, InodeType::from(dev.type_()).into(), mode.into())?;
        inode.set_device_id(dev.id().into()).unwrap();
        Ok(inode)
    }
```